src/python/WMCore/Cache/GenericDataCache.py

```diff
@@ -2,9 +2,11 @@
 import time
 import traceback
 import logging
+from future.utils import with_metaclass
 
+from Utils.Patterns import Singleton
 
-class MemoryCacheStruct(object):
+class MemoryCacheStruct(with_metaclass(Singleton, object)):
```

Comment:
Does it mean there will be only one cache object (MemoryCacheStruct actually) shared among all the components in the agent? And another object shared in ReqMgr?
If the answer is yes, then how we make it thread safe?

Reply:
No it means it will be shared among in one component.  This is used in WorkQueueManager and ErrorHandler. Each component will have its own Singleton object.

Reply:
It is not thread safe.